### PR TITLE
Improvements and 33

### PR DESCRIPTION
**Pull request by MaksyKun**
Briefly said.. a bunch of refactoring and enhancement work. In detail:

# Some related issue-items
#25 Added support for sub-folders in professions
#27 Added example professions on fresh install
#33 Added new browse-options like conditions
- Those work also for recipes so that you can hook into Fabled, mcMMO, Jobs & AuraSkills

# Added an Ingame-Editor
This is probably the biggest change since its a fully new feature. You can edit the browse.yml and each profession individually in the ingame editor. The details are found in my documentation under [EditorUsage](https://github.com/magemonkeystudio/fusion/blob/improvements_and_33/doc/EditorUsage.md).

## Further changes
I did refactor some code that was hard to work with or outdated for future improvements. Most of it is related to the GUIs which were nested and lead to very fast bugs on implementing new features.

### CODE REVIEW

File: src/main/java/studio/magemonkey/fusion/util/CharacterIncrementer.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        // Test the getNextChar method
+        char testChar1 = '0'; // Should return '1'
+        char testChar2 = '9'; // Should return 'a'
+        char testChar3 = 'z'; // Should return 'A'
+        char testChar4 = 'Z'; // Should return 'Z' (no wrap around)
+
+        System.out.println("Next char for '0': " + getNextChar(testChar1));
+        System.out.println("Next char for '9': " + getNextChar(testChar2));
+        System.out.println("Next char for 'z': " + getNextChar(testChar3));
+        System.out.println("Next char for 'Z': " + getNextChar(testChar4));
```
Comment (Travja) [inline src/main/java/studio/magemonkey/fusion/util/CharacterIncrementer.java:31]: This logic would go great in a unit test

File: doc/EditorUsage.md [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+    - Usage:
+        - Left Click: Increase the cost by 1
+        - Right Click: Decrease the cost by 1
+        - Shift + Click: Manage each by 100
+          <br><br>
+- **Recipes**: The recipes gui of the profession. Here you will add and edit recipes in detail.
+    - Shown as: Map
+    - Usage: Click on the item to navigate to the [Recipes Editor](#using-recipe-editor)
+      <br><br>
+- **Use Categories**: Rather you want to split the recipes guis of `/craft use <profession` into sub-categories or not.
```
[the remark this answers is not in the document]
Comment (MaksyKun) [reply doc/EditorUsage.md]: Yes, thanks i added it.

File: src/main/java/studio/magemonkey/fusion/util/CharacterIncrementer.java (hunk 1 above)
Comment (MaksyKun) [reply src/main/java/studio/magemonkey/fusion/util/CharacterIncrementer.java]: Made a setup for unit testing now which includes this test.

**Comment by MaksyKun** [comment]
resolved the conversations so far 👍

**Review by Travja — APPROVED** [review]

**Pull request merged**